packages/hardhat-core/src/internal/core/config/config-loading.ts

```diff
@@ -220,6 +233,25 @@ function readPackageJson(packageName: string): PackageJson | undefined {
   }
 }
 
+function checkEmptyConfig(
+  userConfig: any,
+  {
+    showSolidityConfigWarnings = false,
```

Comment:
I don't think it's necessary to have default values here, especially because it's a private function that's only called once.